### DISCUSSION TITLE
Question regarding interactive() on Linux

### DISCUSSION THREAD

**Discussion by andama** [marcomusy/vedo#579] [Q&A] [answered]
Hi!

Im running multiple plotters in a program by calling the function below multiple times: (`VedoPlotWindow` is not really a window, just a class)
```python
import vedo as v
v.settings.immediateRendering = False
v.settings.allowInteraction = True
def show(mesh):
    #Disregard these 3 rows
    app = init_app() 
    plot_window = VedoPlotWindow.instance().plot_window
    fig = plot_window.fig

    opts = dict(axes=4, interactive=False, new=True, bg='k', title=f'Figure {fig+1} - CALFEM vedo visualization tool')
    plt = v.show(mesh, **opts)
```
Then I'm finally rendering the scenes using:
```python
v.interactive()
```

On macOS, everything works as it should, I can interact with all plotters in separate windows, on Linux however, I can only interact with the last created one. I tried running `vedo -r multiwindows2` on both platforms to test (because I'm basically doing the same thing as in this example). Same thing happens, I can only interact with the last created renderer in Linux, but all of them on macOS.

The following is the printouts from `vedo --info`:

macOS:
```
vedo version      : 2022.0.1   https://vedo.embl.es
vtk version       : 9.0.3
python version    : 3.8.8 (default, Apr 13 2021, 12:59:45) [Clang 10.0.0 ]
python interpreter: /Users/andama/opt/anaconda3/bin/python
vedo installation : /Users/andama/opt/anaconda3/lib/python3.8/site-packages/vedo
system            : Darwin 21.1.0 posix x86_64
```

Linux:
```
vedo version      : 2022.0.1   https://vedo.embl.es
vtk version       : 9.0.3
python version    : 3.8.8 (default, Apr 13 2021, 19:58:26) [GCC 7.3.0]
python interpreter: /home/andama/anaconda3/bin/python
vedo installation : /home/andama/anaconda3/lib/python3.8/site-packages/vedo
system            : Linux 5.4.0-92-generic posix x86_64
```

Is this a known issue, or am I potentially doing something wrong? I've tried all combinations of the `immediateRendering`- and `allowInteraction`-settings.

Also, the important thing is for it to work on Windows in the end. I can't try it on that platform right now, but is it likely to work on Windows if it's working on macOS? For instance, is the `multiwindows2` example working on Windows?

**Comment by marcomusy** [comment]
Hi Andreas, unfortunately this is a new "feature" of the upstream VTK and doesn't happen with versions <=8.1 (if I remember correctly).
So one option would be to use a slightly older vtk version.
I never used windows OS but I can find a machine to try out the `multiwindows2` example at least.

**Reply by andama** [reply]
I see, it's very weird that's it's only happening on one system when both are running VTK 9.0.3 in that case. I'll note the "feature" and VTK versions to try regardless.

If you can try the example on a windows machine and comment on whether interaction works for all windows that would be very helpful. But don't go out of your way, I'll do it myself on Monday regardless ;)

Thank you for the response Marco!

**Comment by marcomusy** [comment, marked as answer]
Now I remember I submitted this same issue to the vtk forum and they say the fixed it... but it looks like the bug is still there...
https://discourse.vtk.org/t/vtk9-behavior-of-vtkrenderwindowinteractor/5587/10

on the other hand on windows it should work ok :)
